src/pgp.rs

```diff
@@ -97,18 +97,29 @@ pub fn create_keypair(addr: impl AsRef<str>) -> Option<(Key, Key)> {
     Some((Key::Public(public_key), Key::Secret(private_key)))
 }
 
+/// Select subkey of the public key to use for encryption.
+///
+/// Currently the first subkey is selected.
+fn select_pk_for_encryption(key: &SignedPublicKey) -> Option<&SignedPublicSubKey> {
+    key.public_subkeys.iter().find(|_k|
```

Comment:
this does not have to be a subkey, the top level key might as well be a signing key

Reply:
I wrote about this above:
> Also maybe consider trying the primary key too, because it has `SignedPublicKey` type and subkeys are of `SignedPublicSubKey` type.

It can be solved by introducing a function to convert `SignedPublicKey` to `SignedPublicSubKey` (just drop its subkeys?), making wrapper type or some dynamic dispatch trickery.

How about adding `From` trait for `SignedPublicKey` to `SignedPublicSubKey` conversion to rPGP?

In any case, until rPGP has functions to determine whether the key can be used for encryption (https://github.com/rpgp/rpgp/pull/69 and maybe adding some API to check KeyFlags for intended usage), it does not make sense to try to use primary key, maybe only in the case when there are no subkeys.